### PR TITLE
Add AGAT tool

### PR DESCRIPTION
**Pull request by gallardoalba**
AGAT has the power to check, fix, pad missing information (features/attributes) of any kind of GTF and GFF to create complete, sorted and standardised gff3 format. Over the years it has been enriched by many many tools to perform just about any tasks that is possible related to GTF/GFF format files (sanitizing, conversions, merging, modifying, filtering, FASTA sequence extraction, adding information, etc). Comparing to other methods AGAT is robust to even the most despicable GTF/GFF files.

Source: https://github.com/NBISweden/AGAT

### CODE REVIEW

**Review by bgruening — APPROVED** [review]

**Review by bgruening — COMMENTED** [review]
maybe compress the genome.fasta?

File: tools/agat/macros.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <xml name="citations">
+        <citations>
+            <citation type="doi">10.5281/zenodo.3552717</citation>
+        </citations>
+    </xml>
+    <xml name="ANNOTATION_INPUT" token_format="gff,gtf,gff3,gff3.gz">
+        <param argument="--gff" type="data" format="@FORMAT@" label="Annotation file" help="Input GTF/GFF file" />
+    </xml>
+    <xml name="REFERENCE_FASTA">
+        <param argument="--fasta" type="data" format="fasta" label="Reference FASTA file" help="Input FASTA file" />
```
Comment (bgruening) [inline tools/agat/macros.xml:24]: should that not use the all_fasta loc file as well?

Comment (gallardoalba) [reply tools/agat/macros.xml:24]: Sorry, I do not understand what do you mean.

Comment (bgruening) [reply tools/agat/macros.xml:24]: Its a reference genome. Should it not be use the Galaxy shipped reference genomes - at least optionally?

Comment (gallardoalba) [reply tools/agat/macros.xml:24]: > Its a reference genome. Should it not be use the Galaxy shipped reference genomes - at least optionally?

Ah, yes, I'll include also the references from the databases.

**Comment by gallardoalba** [comment]
Do you think any of the missing functions could be interesting for you @abretaud? I can include them if needed.

**Pull request merged**